testonly/hammer/hammer.go

```diff
@@ -252,6 +253,35 @@ func HitMap(ctx context.Context, cfg MapConfig) error {
 	return firstErr
 }
 
+// mapWorker allows each thread to have its own PRNG, which makes it easy to
+// reason about the behaviour of each thread being internally consistent and
+// reproducible given a seed. Obviously threads that depend on the state of
+// the map may still have non-deterministic behaviour even when replayed from
+// the same seed.
```

Comment:
So far this comment looks like the only purpose of `mapWorker` is this PRNG thing. Consider framing the comment in terms of what the worker does.

Reply:
Redone this comment. I wrote this first mostly to justify it to myself I guess :-)